compiler/rustc_mir_transform/src/simplify.rs

```diff
@@ -66,8 +66,8 @@ impl SimplifyCfg {
     }
 }
 
-pub(super) fn simplify_cfg(body: &mut Body<'_>) {
-    CfgSimplifier::new(body).simplify();
+pub(super) fn simplify_cfg<'tcx>(tcx: TyCtxt<'tcx>, body: &mut Body<'tcx>) {
```

Comment:
Can you add a big fat warning somewhere prominently in this file that this optimization is one of the few that runs already on analysis MIR as so we must be extremely careful since changes here can affect which programs even compile in an insta-stable way?

Reply:
Do we *need* it to be running on analysis MIR?

Reply:
it's really nice for *real* trivial terminators like gotos and empty bb's